Docs update

ShahFragcolor opened the pull request:
<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
docs-update-1

- updated shards syntax in docs
-deleted repetition and updates some wrong docs

ShahFragcolor committed:
docs-update-2

- edited Remove details
- added sections from Fbl docs

sinkingsugar commented on docs/details/shards/Assert/IsAlmost.md:
```diff
@@ -1,4 +1,6 @@
- For non-decimal values (e.g. `Int`), `:Epsilon` is ignored and this shard acts as [`Assert.Is`](../Is).
+Checks whether the input value lies within the range defined by the `Threshold:` parameter around the specified `Value:`. The shard returns true if the input is within this range (exclusive of the lower limit and inclusive of the upper limit), and false otherwise.
```
`:`

sinkingsugar commented on docs/details/shards/Assert/IsNot.md:
```diff
@@ -1,8 +1,6 @@
- In this shard it's the inequality that's asserted. This means the assertion is considered successful if input to this shard does **not** match the `:Value` parameter of this shard.
+Returns true if the input is not equal to the `Value:` parameter and false otherwise. The shard also is type sensitive (e.g., 1 | IsNot(1.0) will return true).
```
`:`

sinkingsugar commented on docs/details/shards/General/Clamp.md:
````diff
@@ -0,0 +1,15 @@
+Consider the following example:
+``` shards
+@i2(2 5)
+  Clamp(Min: @i2(3 3) Max: @i2(4 4))
````
extra spaces

sinkingsugar commented on docs/details/shards/General/Const.md:
```diff
@@ -1,11 +1,7 @@
-
 A constant value declared with `Const` is un-named i.e., it is not assigned to any variable or allocated any alias. Hence it cannot be invoked or referred to later. To create named constants see [`Ref`](../Ref).
 
 You can even skip this shard and pass the constant value directly but internally it will be translated to a `Const`shard that outputs this constant value. However, it's good practice to use this keyword in while passing constants in Shards programs.
```
> However, it's good practice to use this keyword in while passing constants in Shards programs.

not really why? Maybe some old text?

sinkingsugar commented on docs/details/shards/General/Const.md:
```diff
@@ -1,11 +1,7 @@
-`Const` declares a constant value (of any data type) by passing it into the parameter `:Value`. Such a value is usually declared for use as input in other shards.
-
 A constant value declared with `Const` is un-named i.e., it is not assigned to any variable or allocated any alias. Hence it cannot be invoked or referred to later. To create named constants see [`Ref`](../Ref).
```
This is a bit confusing, you still need a const to use ref anyway :D

sinkingsugar commented on docs/details/shards/General/Do.md:
```diff
@@ -0,0 +1,5 @@
+
+Child wire is scheduled and executes inline. Any pauses on the child wire will also pause the parent wire.
+
+Child wire is scheduled on the same mesh as the parent wire.
```
child wire is not scheduled, it's executed in line, not scheduled.

sinkingsugar commented on docs/details/shards/General/DoMany.md:
```diff
@@ -0,0 +1,5 @@
+
+Child wires are scheduled and executes inline. The parent wire will execute all child wires sequentially, it will only continue its execution once all child wires have finished.
+
+Child wires are scheduled on the same mesh as the parent wire.
```
also not scheduled

sinkingsugar commented on docs/details/shards/General/DoMany.md:
```diff
@@ -0,0 +1,5 @@
+All child wires scheduled inherits and uses the context variables of the parent wire. Any changes to said variables will also be reflected on the parent wire and subsequent child wires scheduled.
```
not scheduled, but wires are cloned, meaning each input will have its own context

sinkingsugar commented on docs/details/shards/General/Erase.md:
```diff
@@ -1,14 +1,8 @@
+The `Indices` parameter specifies which elements or keys to remove from the target sequence or table. For example, `Erase(Indices: [0 1] Name: sequence)` removes the elements at indices 0 and 1 from a sequence, while `Erase(Indices: ["key1" "key2"] Name: table) deletes the keys "key1" and "key2" (and their values)` from a table.
 
-For a sequence, this shard expects the index (or a sequence of indices in descending order) of the element(s) to be erased, followed by the name of the sequence variable in the `:Name` parameter.
+The `Key:` parameter identifies which key contains the value to be modified. The value associated with this key must itself be a table or a sequence.
```
`:`

sinkingsugar commented on docs/details/shards/General/Expand.md:
```diff
@@ -0,0 +1,5 @@
+All child wires scheduled makes copies of the parent wire's context variables. Any changes to said variables will not be reflected on the parent wire or other sibling wires scheduled using this shard.
+
+Child wires are scheduled and executes inline. However, all child wires scheduled using this shard will execute in paralell. This shard however, will wait for all child wires to finish executing before it will continue its execution. Therefore any pauses on any child wire will also pause the parent wire.
```
A bit confusing, wires are either scheduled or executed inline.. in this case it's scheduled, possibly on different threads too. The shard will output when all wires finished.

sinkingsugar commented on docs/details/shards/General/FromJson.md:
```diff
@@ -0,0 +1 @@
+If the input JSON string is a Shards-specific JSON string and not a standard JSON string, if the `Pure` parameter is set to true, the shard will lose the original type information and always return a table.
```
not really, it can output (not return btw) anything. when Pure is true, it will threat the input as Pure JSON, not Shards flavored JSON (which includes shards metadata)

sinkingsugar commented on docs/details/shards/General/Is.md:
```diff
@@ -1,6 +1,4 @@
-This shard compares the input to its `:Value` parameter and outputs `true` if they are equal or same, else outputs `false`.
-
-If the input and `:Value` parameter have different data types they will be assessed as inequal by default even if they are numerically equal (for example `int 5` is not equal to `float 5.0`).
+This shard also is type sensitive (e.g., 1 | Is(1.0) will return false).
```
I think `1 | Is(1.0)` will now fail!

sinkingsugar commented on docs/details/shards/General/PopFront.md:
```diff
@@ -1,10 +1,6 @@
-`PopFront` drops (removes) the first element of the sequence variable that has been passed to in the `:Name` parameter and makes it available to the next shard as its input.
+This shard works on both sequences and tables. Parameter `Key:` applies only to tables.
```
`:`

sinkingsugar commented on docs/details/shards/General/Reduce.md:
```diff
@@ -0,0 +1,15 @@
+This shard is able to use a special temporary variable `$0` which acts as an Accumulator which will hold the intermediate result during the reduction process.
+
+For example consider the following: `[1 2 3 4 5] | Reduce(Math.Subtract($0))`
```
does this work? btw this Shard is now `Fold` `Reduce` is an alias to Fold

sinkingsugar commented on docs/details/shards/General/Ref.md:
```diff
@@ -1,10 +1,8 @@
-`Ref` creates an immutable variable and assigns a constant value to it. Once created this variable cannot be changed.
+The name of the variable comes from the `Name` parameter and the constant value comes from the input. The type of input controls the kind of variable that will created: numeric input creates numeric variable, string input creates string variable, and sequence input would create a sequence variable.
```
"that will created:` ? something is off

sinkingsugar commented on docs/details/shards/General/Remove.md:
```diff
@@ -1,20 +1,16 @@
-`Remove` removes all the elements of the `:Name` parameter sequence that matches the condition laid out in the `:Predicate` parameter.
-
-It can also take these condition-matched indices (from the `:From` sequence) and remove the corresponding elements from a joined sequence (passed via the `:Join` parameter). Remember, `Remove` doesn't apply the `:Predicate` conditions to the joined sequence, but removes corresponding elements from it based on `:Predicate` matched indices of the main sequence. For this to work both the sequences m ...
+This shard can also remove elements with the corresponding indices from a joined sequence (passed via the `Join:` parameter). Remember, `Remove` doesn't apply the `Predicate:` conditions to the joined sequence, but only removes corresponding elements. For this to work both the sequences must have the same length.
```
`:`

sinkingsugar commented on docs/details/shards/General/Remove.md:
```diff
@@ -1,20 +1,16 @@
     In this case the operation is deletion of selected elements (selected rows) from one sequence (table) leading to deletion of corresponding elements (connected rows) of the joined sequence (joined table).
 
-The `:Predicate` parameter can take any conditional/logical expression or combination of shards that will result in assertion that can be tested on the sequence elements.
+The `Predicate:` parameter can take any conditional/logical expression or combination of shards that will result in assertion that can be tested on the sequence elements.
```
`:`

sinkingsugar commented on docs/details/shards/General/Remove.md:
```diff
@@ -1,20 +1,16 @@
+The `Predicate:` parameter can take any conditional/logical expression or combination of shards that will result in assertion that can be tested on the sequence elements.
 
-The `:Unordered` parameter can be set to `true` if you need to make the removal process faster, but then the order of the remaining elements in the resulting sequence elements may not be preserved. By default, this order is preserved.
+The `Unordered:` parameter can be set to `true` if you need to make the removal process faster, but then the order of the remaining elements in the resulting sequence elements may not be preserved. By default, this order is preserved.
```
`:`

sinkingsugar commented on docs/details/shards/General/Table.md:
```diff
@@ -1,10 +1,8 @@
-`Table` creates an empty table with or without a specified key (via the `:Key` parameter). The created table name is defined in the `:Name` parameter.
+`Table` creates an empty table with or without a specified key (via the `Key` parameter). The created table name is defined in the `:Name` parameter.
```
`:Name`

sinkingsugar commented on docs/details/shards/General/Update.md:
```diff
@@ -1,17 +1,12 @@
-
-The name of the variable comes from the `:Name` parameter and the update value comes from the input. 
 
 `Update` overwrites string, numeric, and sequence variables with the new value (coming from input). However, for sequences, it cannot update a sequence at the level of elements (i.e., add elements, remove elements, change element order, etc.), so it overwrites the whole sequence with whatever you've passed in the input field.
```
not sure if `overwrite` is a good word here.. just `writes` maybe? or changes?

sinkingsugar commented on docs/details/shards/GFX/Drawable.md:
````diff
@@ -18,17 +18,17 @@ Shader parameters and the drawable transform can be set as either constant param
 Constant parameters are passed through the input table. They are read and stay the same until this function is called again.
 
 ```clojure
````
clojure? use ```shards, fix if more clojure is found

sinkingsugar commented on docs/details/shards/GFX/Material.md:
````diff
@@ -1,2 +1,13 @@
+The `Params` parameter can accept a table with keys representing the shader parameters for the material and their respective values. For example:
+  ``` shards
````
extra space after triple ticks

sinkingsugar commented on docs/details/shards/Http/Delete.md:
````diff
@@ -0,0 +1,22 @@
+If the output of the shard is a table, it will have the following format:
+```shards
+{
+  status: Int,
````
use proper shards enums, e.g. Type::Int

sinkingsugar commented on docs/details/shards/Http/Patch.md:
````diff
@@ -0,0 +1,24 @@
+- **bytes** : The bytes are attached to the body as is.
+
+If the output of the shard is a table, it will have the following format:
+```shards
````
use proper shards synthax like under (Post)

sinkingsugar commented on docs/details/shards/Http/Put.md:
````diff
@@ -0,0 +1,24 @@
+If the output of the shard is a table, it will have the following format:
+```shards
+{
+  status: Int,
````
shards synthax

sinkingsugar replied:
remove commas, format it too

sinkingsugar commented on shards/modules/core/core.cpp:
```diff
@@ -523,7 +523,7 @@ struct XPendBase {
 };
 
 struct AppendTo : public XPendBase {
```
`:`

ShahFragcolor committed:
changes based on comments

- changed reduce to fold
- also changed the help documentation in fold shard to reflect current behaviour

ShahFragcolor committed:
more changes based on comments

sinkingsugar commented on docs/details/shards/Assert/Is.md:
```diff
@@ -1,10 +1,6 @@
-If the assertion is satisfied (i.e., the input is equal to or same as the `:Value` parameter) the program will is allowed to continue (control passes to the next shard), irrespective of the `:Break` parameter .
+Returns true if the input is equal to the `Value:` parameter and false otherwise. The shard also is type sensitive (e.g., 1 | Is(1.0) will return false).
```
`:`

sinkingsugar commented on docs/details/shards/Assert/Is.md:
```diff
@@ -1,10 +1,6 @@
-- logs an assertion validation error but continues running (control passes to the next wire scheduled on the mesh).
-
-Since this shard can precisely control the conditions under which a program is allowed to run or is to be aborted, it's effective for writing (inline) unit test cases with it.
+If the `Break` parameter is set to true - logs an assertion validation error but continues running the programme.
```
`programme` and this is wrong, with break will either break if debugger attached or crash

sinkingsugar commented on docs/details/shards/Assert/IsAlmost.md:
```diff
@@ -1,4 +1,6 @@
- For non-decimal values (e.g. `Int`), `:Epsilon` is ignored and this shard acts as [`Assert.Is`](../Is).
+Checks whether the input value lies within the range defined by the `Threshold` parameter around the specified `Value`. The shard returns true if the input is within this range (exclusive of the lower limit and inclusive of the upper limit), and false otherwise.
+
+If the `Break` parameter is set to true - logs an assertion validation error but continues running the programme.
```
wrong like above

sinkingsugar commented on docs/details/shards/Assert/IsNot.md:
```diff
@@ -1,8 +1,6 @@
-The rest of behaviour of this shard (including the impact of the value of the `:Break` parameter) is identical to [`Assert.Is`](../Is).
-
-Just like [`Assert.Is`](../Is), this shard too can be used for writing (inline) unit test cases.
+If the `Break` parameter is set to true - logs an assertion validation error but continues running the programme.
```
wrong

sinkingsugar commented on docs/details/shards/General/Assoc.md:
```diff
@@ -1,3 +1,3 @@
-This shard can be used to update specific member elements within a sequence or a table with new values.
+The input sequence should contain pairs of elements: the first in each pair specifies the index or key to update, and the second provides the new value. Therefore, the input sequence must always have an even number of elements.
```
`elements:` ?

sinkingsugar replied:
Also weird flow

sinkingsugar commented on docs/details/shards/General/Const.md:
```diff
@@ -1,12 +1,6 @@
-You can also use `(Const nil)` to overwrite/nullify (since `nil` stands for null data) the input to the subsequent shard, if required (see the last code sample).
-
-Input field is ignored and the output of this shard is the constant value defined by it.
+You can even skip this shard and pass the constant value directly but internally it will be translated to a `Const`shard that outputs this constant value.
```
missing space after `Const`

sinkingsugar commented on docs/details/shards/General/Expand.md:
```diff
@@ -1,5 +1,5 @@
 All child wires scheduled makes copies of the parent wire's context variables. Any changes to said variables will not be reflected on the parent wire or other sibling wires scheduled using this shard.
 
-Child wires are scheduled and executes inline. However, all child wires scheduled using this shard will execute in paralell. This shard however, will wait for all child wires to finish executing before it will continue its execution. Therefore any pauses on any child wire will also pause the parent wire.
+All child wires scheduled using this shard will execute in paralell. This shard however, will wait for all child wires to finish executing before it will continue its execution. Therefore any pauses on any child wire will also pause the parent wire.
```
`parallel` (you should use a spellcheck vscode extension 👍 )

sinkingsugar commented on docs/details/shards/GFX/Drawable.md:
````diff
@@ -17,15 +17,15 @@ Shader parameters and the drawable transform can be set as either constant param
 Constant parameters are passed through the input table. They are read and stay the same until this function is called again.
 
-```clojure
+```shard
````
eh.. `shards`

sinkingsugar commented on docs/details/shards/GFX/Drawable.md:
````diff
@@ -17,15 +17,15 @@ Shader parameters and the drawable transform can be set as either constant param
 Dynamic parameters are passed as parameters to the shard. You can set the same fields as you can when setting constant parameters, with the exception of the mesh. The variables are referenced by the `Drawable` so that changes in their value will be reflect in the rendered result.
 
-```clojure
+```shard
````
see above

sinkingsugar commented on docs/details/shards/GFX/Drawable.md:
````diff
@@ -35,7 +35,7 @@ Dynamic parameters are passed as parameters to the shard. You can set the same f
 To avoid re-creating Drawables for objects with minor or no changes, you should set up Drawables from within a [Setup](../../General/Once) block and pass their dynamic parameters to the shard.
 
-```clojure
+```shard
````
...

sinkingsugar commented on docs/details/shards/GFX/glTF.md:
````diff
@@ -25,7 +25,7 @@ The input table can include keys representing nodes to modify. The keys can be a
   - "params": A table of node-specific parameters (if applicable)
 
-```clojure
+```shard
````
...

ShahFragcolor committed:
updates based on comments
